instance/models/mixins/openedx_site_configuration.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2020 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance app model mixins - SiteConfiguration parameters
+"""
+
+import yaml
+
+from django.db import models
+
+
+class OpenEdXSiteConfigurationMixin(models.Model):
+    """
+    Mixin to provide the SiteConfiguration parameters to be set on the instance.
+    """
+    class Meta:
+        abstract = True
+
+    def get_site_configuration_settings(self):
+        """
+        Return the ansible variables to set the SiteConfiguration parameters.
+        """
+        site_configuration_settings = {}
+
+        if self.static_templates_overrides:
+            static_templates_overrides = {k: v for k, v in self.static_templates_overrides.items() if k != 'version'}
+            if static_templates_overrides:
+                site_configuration_settings.update(static_templates_overrides)
+
+        if site_configuration_settings:
+            return yaml.dump(
+                {'EDXAPP_SITE_CONFIGURATION': [{'values': site_configuration_settings}]},
+                default_flow_style=False
+            )
+        else:
+            return ''
```

Comment:
Shouldn't we return `{}` instead of an empty string? This might cause issues when merging the dictionaries, no?

Reply:
@giovannicimolin, the return statement inside the `if` block above returns the string containing the rendered variables. So the empty string `''` is consistent with that. If we have to merge the `SiteConfiguration` variables from different sources - for example, there will be one needed for the home page hero text customizastion, we have the `yaml_merge` function [here](https://github.com/open-craft/opencraft/blob/master/instance/ansible.py#L57) that can be used like it has been in a few existing places.